### PR TITLE
fix no more available PCI slots issue in virsh_attach_detach_interface

### PR DESCRIPTION
**Pull request by chunfuwen**
In some casees, PCI slots may be taken up,so it may throw no more avaiable PCI slots error
if attach second interface.

Fix it by cleaning up pci controller except preallocated index=0 controller
Signed-off-by: chunfuwen <chwen@redhat.com>

### CODE REVIEW

**Comment by chunfuwen** [comment]
paste piece of error log:
```
2018-06-03 03:58:22,725 virsh            L1395 DEBUG| Undefine VM avocado-vt-vm1
2018-06-03 03:58:22,766 virsh            L1379 DEBUG| Define VM from /var/tmp/xml_utils_temp_ltoq2qkc.xml
2018-06-03 03:58:22,805 stacktrace       L0041 ERROR| 
2018-06-03 03:58:22,806 stacktrace       L0044 ERROR| Reproduced traceback from: /var/tmp/libvirt-ci/runtest/avocado-vt/avocado-vt/avocado_vt/test.py:434
2018-06-03 03:58:22,807 stacktrace       L0047 ERROR| Traceback (most recent call last):
2018-06-03 03:58:22,808 stacktrace       L0047 ERROR|   File "/var/lib/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-libvirt/libvirt/tests/src/virsh_cmd/domain/virsh_attach_detach_interface.py", line 316, in run
2018-06-03 03:58:22,808 stacktrace       L0047 ERROR|     test.fail(result_info)
2018-06-03 03:58:22,808 stacktrace       L0047 ERROR|   File "/var/tmp/libvirt-ci/runtest/avocado-vt/avocado/avocado/core/test.py", line 1052, in fail
2018-06-03 03:58:22,808 stacktrace       L0047 ERROR|     raise exceptions.TestFail(message)
2018-06-03 03:58:22,808 stacktrace       L0047 ERROR| avocado.core.exceptions.TestFail: ['Attach Failed: error: Failed to attach interface\nerror: internal error: No more available PCI slots\n']
2018-06-03 03:58:22,808 stacktrace       L0048 ERROR| 
2018-06-03 03:58:22,915 virt_vm          L1093 DEBUG| Attempting to log into 'vm2' (timeout 240s)
```

**Comment by bssrikanth** [comment]
@chunfuwen error no more pci slots araise when user/test has consumed all available pci solts for some purpose. Deleting all pci controllers apart from default one, seems to be unsafe. We should be checking the tests which are consuming it and not releasing/deleting them at end of the test. I would think that would be better solution.

**Comment by chunfuwen** [comment]
@bssrikanth , that's true. But actually we are running this issue in the queue with multiple cases, so it is quite time-consuming and painful to locate which case may leads to this.

File: libvirt/tests/src/virsh_cmd/domain/virsh_attach_detach_interface.py [hunk 1]
```diff
@@ -246,6 +246,21 @@ def run(test, params, env):
     logging.info("iface_format is %s", iface_format)
 
     try:
+        # Delete all controller devices except index=0 to allow available PCI slots.
```
Comment (balamuruhans) [inline libvirt/tests/src/virsh_cmd/domain/virsh_attach_detach_interface.py]: I would agree with @bssrikanth, The clean up of the testcases should take care of deleting the pci controllers after test. let us do the right thing eventhough it is time consuming.

**Comment by iccaszhulili** [comment]
rh-caselink retest please

**Pull request closed without merging**